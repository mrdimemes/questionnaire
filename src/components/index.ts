export { App } from "./App";

export { Header } from "./header";

export { PaginationBar } from "./pagination";

export { TagComponent } from "./tags";
export { TagGallery } from "./tags";
export { TagBar } from "./tags";
export { AddTagWidget } from "./tags";

export { QuestionnaireGallery } from "./questionnaires/gallery";
export { QuestionnaireComponent } from "./questionnaires/questionnaire";
export { QuestionnaireEditor } from "./questionnaires/edit";

export { Button } from "./UI/Button";
export { Input } from "./UI/inputs";
export { CheckboxInput } from "./UI/inputs";
export { RadioInput } from "./UI/inputs";
export { TextInput } from "./UI/inputs";
export { EmailInput } from "./UI/inputs";
export { PasswordInput } from "./UI/inputs";
export { NameInput } from "./UI/inputs";
export { TextArea } from "./UI/inputs";
export { Select } from "./UI/inputs";

export { AuthForm } from "./AuthForm";

export { LoadingSpinner } from "./LoadingSpinner";