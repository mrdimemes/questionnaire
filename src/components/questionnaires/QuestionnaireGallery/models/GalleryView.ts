export enum GalleryView {
  Rows = "ROWS",
  Plates = "PLATES",
};