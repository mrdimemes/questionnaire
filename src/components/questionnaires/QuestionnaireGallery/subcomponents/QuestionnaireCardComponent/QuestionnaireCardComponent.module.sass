@use "src/sass/foundation/colors"
@use "src/sass/foundation/mixins"
@use "src/sass/foundation/breakpoints"

.QuestionnaireCard
  position: relative

.card
  display: block
  padding: 10px
  border-radius: 6px
  cursor: pointer
  height: max-content

.label
  margin-bottom: 0.8em

.adminBar
  position: absolute
  top: 10px
  right: 10px
  display: flex
  column-gap: 0.5em

.icon
  @include mixins.square(10px)
  border: 0
  border-radius: 0
  padding: 0
  transition: background-color 0.3s

.deleteButton
  @include mixins.mask-image("/assets/images/icons/close.svg")

.editLink
  @include mixins.mask-image("/assets/images/icons/edit.svg")

.theme_dark
  & .card:hover
    background-color: colors.$dark-theme-bg-highlight
  & .deleteButton, .editLink
    background-color: colors.$dark-theme-4
    &:hover
      background-color: colors.$dark-theme-error

.theme_light
  & .card:hover
    background-color: colors.$light-theme-bg-highlight
  & .deleteButton, .editLink
    background-color: colors.$gray-1
    &:hover
      background-color: colors.$light-theme-error
