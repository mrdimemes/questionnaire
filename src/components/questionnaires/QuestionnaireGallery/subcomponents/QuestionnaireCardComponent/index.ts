export { default as QuestionnaireCardComponent } from "./QuestionnaireCardComponent";