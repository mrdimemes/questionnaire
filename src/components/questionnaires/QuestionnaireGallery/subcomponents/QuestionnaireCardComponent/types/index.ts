export type {
  QuestionnaireCardComponentProps,
} from "./QuestionnaireCardComponentProps";