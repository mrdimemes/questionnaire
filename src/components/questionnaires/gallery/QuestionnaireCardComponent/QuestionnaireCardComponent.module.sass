@use "src/sass/foundation/colors"
@use "src/sass/foundation/breakpoints"

.cardContainer
  position: relative

.card
  display: block
  padding: 10px
  border-radius: 6px
  cursor: pointer
  height: max-content

.label
  margin-bottom: 0.8em

.theme_dark
  &:hover
    background-color: colors.$dark-theme-bg-highlight

.theme_light
  &:hover
    background-color: colors.$light-theme-bg-highlight

.adminBar
  position: absolute
  top: 10px
  right: 10px
  display: flex
  column-gap: 0.3em

.button
  transition: color 0.3s
  border: 0
  background-color: transparent
  border-radius: 15px
  padding: 3px 10px