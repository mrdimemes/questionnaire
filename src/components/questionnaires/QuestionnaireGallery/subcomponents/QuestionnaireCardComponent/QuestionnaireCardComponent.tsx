import classNames from "classnames";
import { useNavigate } from "react-router-dom";

import { TagBar, Button } from "src/components";
import { withTheme } from "src/HOCs";
import { useUserAdminFlagSelector } from "src/redux/hooks";
import { QuestionnaireService } from "src/services";
import { executeWithConfirmation } from "src/utils/helpers";

import styles from "./QuestionnaireCardComponent.module.sass";

import type { QuestionnaireCardComponentProps } from "./types";


const QuestionnaireCardComponent = ({
  id,
  label,
  tags,
  className,
}: QuestionnaireCardComponentProps,
) => {
  const navigate = useNavigate();
  const isUserAdmin = useUserAdminFlagSelector();

  const removeQuestionnaire = () => {
    executeWithConfirmation(
      "Вы уверены что хотите удалить этот опрос?",
      () => QuestionnaireService.removeQuestionnaire(id),
    );
  };

  return (
    <div
      className={classNames(styles.QuestionnaireCard, className)}
      onClick={() => navigate("/questionnaire/" + id)}
    >

      <h2 className={styles.label}>{label}</h2>
      <TagBar tags={tags} />

      {
        isUserAdmin &&
        <div className={styles.adminBar}>
          <Button
            className={classNames(styles.icon, styles.editButton)}
            onClick={() => navigate("/edit/" + id)}
          />
          <Button
            className={classNames(styles.icon, styles.deleteButton)}
            onClick={removeQuestionnaire} />
        </div>
      }

    </div>
  );
};

export default withTheme(QuestionnaireCardComponent, styles);