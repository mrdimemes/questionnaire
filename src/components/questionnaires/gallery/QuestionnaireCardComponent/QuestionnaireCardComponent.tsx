import classNames from "classnames";
import { Link } from "react-router-dom";
import { useAppSelector } from "src/redux/hooks";
import { getThemeStyle } from "src/redux/slices/themeSlice";
import { TagBar } from "src/components";
import styles from "./QuestionnaireCardComponent.module.sass";

type QuestionnaireCardProps = {
  id: number,
  label: string,
  tags: {
    id: number,
    label: string
  }[]
}

const QuestionnaireCardComponent = ({
  id,
  label,
  tags }: QuestionnaireCardProps
) => {
  const currentTheme = useAppSelector((state) => state.theme.theme);

  return (
    <Link
      to={"/questionnaire/" + id}
      className={classNames(styles.body, getThemeStyle(styles, currentTheme))}
    >
      <h2 className={styles.label}>{label}</h2>
      <TagBar tags={tags} />
    </Link>
  )
}

export default QuestionnaireCardComponent