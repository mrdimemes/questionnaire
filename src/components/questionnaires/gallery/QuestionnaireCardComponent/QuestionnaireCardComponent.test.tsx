import { screen } from "@testing-library/react";
import { FetchStatus } from "src/models";
import { setupStore } from "src/redux";
import { renderWithProvidersAndRouters } from "src/utils/test-utils";

import QuestionnaireCardComponent from "./QuestionnaireCardComponent";

describe("QuestionnaireCard component", () => {

  test("QuestionnaireCard renders", () => {
    const store = setupStore({
      tags: {
        tags: [{ id: 1, label: "TestTag", frequency: 7 }],
        tagsLoadingStatus: FetchStatus.Complete,
      },
    });
    renderWithProvidersAndRouters(
      <QuestionnaireCardComponent id={1} label="QuestionLabel" tags={[1]} />,
      { store },
    );
    expect(screen.getByText(/questionlabel/i)).toBeInTheDocument();
    expect(screen.getByText(/testtag/i)).toBeInTheDocument();
  });

  test("QuestionnaireCard snapshot", () => {
    const store = setupStore({
      tags: {
        tags: [{ id: 1, label: "TestTag", frequency: 7 }],
        tagsLoadingStatus: FetchStatus.Complete,
      },
    });
    const { container } = renderWithProvidersAndRouters(
      <QuestionnaireCardComponent id={1} label="QuestionLabel" tags={[1]} />,
      { store },
    );
    expect(container).toMatchSnapshot();
  });
});