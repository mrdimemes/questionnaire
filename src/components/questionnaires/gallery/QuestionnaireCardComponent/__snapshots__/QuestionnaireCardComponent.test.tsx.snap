// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionnaireCard component QuestionnaireCard snapshot 1`] = `
<div>
  <a
    class="body theme_light"
    href="/questionnaire/1"
  >
    <h2
      class="label"
    >
      QuestionLabel
    </h2>
    <div
      class="body"
    >
       
      <div
        class="body theme_light"
      >
        <div>
          TestTag
        </div>
      </div>
       
    </div>
  </a>
</div>
`;
